src/pal/src/numa/numa.cpp

```diff
@@ -46,11 +46,69 @@ bool g_numaAvailable = false;
 void* numaHandle = nullptr;
 
 #if HAVE_NUMA_H
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
 #define PER_FUNCTION_BLOCK(fn) decltype(fn)* fn##_ptr;
 FOR_ALL_NUMA_FUNCTIONS
 #undef PER_FUNCTION_BLOCK
 #endif // HAVE_NUMA_H
 
+#if defined(__linux__) && HAVE_NUMA_H
+static bool ShouldOpenLibNuma()
+{
+    // This is a simple heuristic to determine if libnuma.so should be opened.  There's
+    // no point in linking and resolving everything in this library if we're running on
+    // a system that's not NUMA-capable.
+    int fd;
+
+    while (true)
+    {
+        fd = open("/sys/devices/system/node/possible", O_RDONLY | O_CLOEXEC);
+
+        if (fd == -1)
+        {
+            if (errno == EINTR)
```

Comment:
I wonder - can this open really fail with EINTR? We have several places in PAL where we use open and we don't check for EINTR anywhere. The Linux doc says that you can get EINTR when the open would be blocked waiting to complete an open of a slow device.

Reply:
Yeah, it can happen. In this case, we can just ignore this error and try use libnuma anyway.